File: surface/widgets/swiftui/components/widgets/navigation/Sidebar.swift
// ============================================================
// Clef Surface SwiftUI Widget — Sidebar
//
// Collapsible sidebar navigation panel with items and sections.
// Uses SwiftUI NavigationSplitView patterns.
// ============================================================

import SwiftUI

// --------------- Types ---------------

struct SidebarItem: Identifiable {
    let id: String
    let label: String
    var icon: String? = nil
    var active: Bool = false
    var badge: String? = nil
}

// --------------- Component ---------------

/// Sidebar view for navigation panel with collapsible items.
///
/// - Parameters:
///   - items: Sidebar navigation items.
///   - collapsed: Whether the sidebar is collapsed.
///   - title: Optional sidebar title.
///   - onSelect: Callback when an item is selected.
///   - onToggle: Callback when collapse state changes.
struct SidebarView: View {
    var items: [SidebarItem]
    @Binding var collapsed: Bool
    var title: String? = nil
    var onSelect: ((SidebarItem) -> Void)? = nil
    var onToggle: (() -> Void)? = nil

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            // Header
            HStack {
                if !collapsed, let title = title {
                    Text(title)
                        .font(.headline)
                        .fontWeight(.semibold)
                }
                Spacer()
                SwiftUI.Button(action: {
                    collapsed.toggle()
                    onToggle?()
                }) {
                    Image(systemName: collapsed ? "sidebar.left" : "sidebar.left")
                        .foregroundColor(.secondary)
                }
                .buttonStyle(.plain)
                .accessibilityLabel(collapsed ? "Expand sidebar" : "Collapse sidebar")
            }
            .padding(.horizontal, 12)
            .padding(.vertical, 8)

            Divider()

            // Items
            ForEach(items) { item in
                SwiftUI.Button(action: {
                    onSelect?(item)
                }) {
                    HStack(spacing: 8) {
                        if let icon = item.icon {
                            Image(systemName: icon)
                                .frame(width: 24)
                                .foregroundColor(item.active ? .accentColor : .secondary)
                        }

                        if !collapsed {
                            Text(item.label)
                                .font(.body)
                                .foregroundColor(item.active ? .accentColor : .primary)

                            Spacer()

                            if let badge = item.badge {
                                Text(badge)
                                    .font(.caption2)
                                    .padding(.horizontal, 6)
                                    .padding(.vertical, 2)
                                    .background(Color.accentColor.opacity(0.1))
                                    .clipShape(Capsule())
                            }
                        }
                    }
                    .padding(.horizontal, 12)
                    .padding(.vertical, 8)
                    .background(item.active ? Color.accentColor.opacity(0.08) : Color.clear)
                }
                .buttonStyle(.plain)
            }

            Spacer()
        }
        .frame(width: collapsed ? 48 : 220)
        .background(Color(.systemBackground))
        .animation(.easeInOut(duration: 0.2), value: collapsed)
    }
}

File: surface/widgets/nativescript/components/widgets/navigation/Sidebar.ts
// ============================================================
// Clef Surface NativeScript Widget — Sidebar
//
// Collapsible sidebar navigation with groups and items.
// ============================================================

import { StackLayout, Label, Button } from '@nativescript/core';

export interface SidebarItem { id: string; label: string; icon?: string; href?: string; badge?: string; }
export interface SidebarGroup { label: string; items: SidebarItem[]; }

export interface SidebarProps {
  groups?: SidebarGroup[];
  items?: SidebarItem[];
  activeId?: string;
  collapsed?: boolean;
  collapsible?: boolean;
  onNavigate?: (id: string) => void;
  onCollapsedChange?: (collapsed: boolean) => void;
}

export function createSidebar(props: SidebarProps): StackLayout {
  const {
    groups = [], items = [], activeId, collapsed = false,
    collapsible = true, onNavigate, onCollapsedChange,
  } = props;

  const container = new StackLayout();
  container.className = 'clef-widget-sidebar';
  container.accessibilityRole = 'navigation';

  if (collapsible) {
    const toggleBtn = new Button();
    toggleBtn.text = collapsed ? '\u25B6' : '\u25C0';
    toggleBtn.accessibilityLabel = collapsed ? 'Expand sidebar' : 'Collapse sidebar';
    toggleBtn.on('tap', () => onCollapsedChange?.(!collapsed));
    container.addChild(toggleBtn);
  }

  const renderItems = (sidebarItems) => {
    for (const item of sidebarItems) {
      const row = new StackLayout();
      row.orientation = 'horizontal';
      row.padding = '8 12';
      row.className = item.id === activeId ? 'clef-sidebar-item-active' : 'clef-sidebar-item';
      if (item.icon && !collapsed) {
        const icon = new Label();
        icon.text = item.icon;
        icon.marginRight = 8;
        row.addChild(icon);
      }
      if (!collapsed) {
        const lbl = new Label();
        lbl.text = item.label;
        row.addChild(lbl);
      }
      row.on('tap', () => onNavigate?.(item.id));
      container.addChild(row);
    }
  };

  for (const group of groups) {
    if (!collapsed) {
      const groupLabel = new Label();
      groupLabel.text = group.label;
      groupLabel.fontWeight = 'bold';
      groupLabel.opacity = 0.6;
      groupLabel.padding = '8 12 4 12';
      groupLabel.fontSize = 11;
      container.addChild(groupLabel);
    }
    renderItems(group.items);
  }
  renderItems(items);

  return container;
}

export default createSidebar;
